sunpy/io/srs.py

```diff
@@ -0,0 +1,124 @@
+import collections, calendar, re
+
+__all__ = ['read']
+
+def read(filepath):
+    """
+    Method for reading an SRS File.
+    """
+    File = open(filepath, 'r')
+    lines = list()
+    for line in File:
+        arr = line.split()
+        store = list()
+        for i in range(0, min(len(arr), 8)):
+            store.append(arr[i])
+        lines.append(store)
+
+    #Store table meta-data. Only first two lines
+    #of SRS text file store the meta-data.
+    meta_data = collections.OrderedDict()
+    regex = re.compile('[^a-zA-Z]')
+    try:
+        meta_data[regex.sub('', lines[0][0])] = lines[0][1]
+        meta_data[regex.sub('', lines[1][0])] = ' '.join(lines[1][1:])
+    except IndexError:
+        print ("SRS meta-data is not available.")
+
+    #Problem: An array of strings. We need to extract
+    #three tables from these I, IA and II.
+
+    table = list() #This table holds the three tables which we would later merge.
+    indices = list()
+    
+    for i in range(0, len(lines)):
+        if (lines[i][0][0] == 'I'):
+            indices.append(i)
+    indices.append(len(lines))
+    indices.sort()
+    
+    for i in range(0, len(indices) - 1):
+        cols = lines[indices[i]+1]
+        temp_table = Table(names = cols, dtype=['object_']*len(cols))
+        #If a table is empty, we are not adding it.
+        for j in range(indices[i]+2, indices[i+1]):
+            temp_string = lines[j]
+            temp_array = temp_string
+            if (len(temp_array) == len(cols)):
+                temp_table.add_row(temp_array)
+        #Make the table, data-type aware while
+        #you're building it.
+        #If it is empty, don't do anything
+        #Just add the empty table as it is.
+        if len(temp_table)>0:
+            for cols in temp_table.columns.values():
+                #Make the table columns unit-aware. First convert string to
+                #floats, ones which can be converted that is.
+                try:
+                    column = temp_table[cols.name].astype(float)
+                    temp_table.replace_column(cols.name, column)
+                except ValueError:
+                    pass
+        table.append(temp_table)
+
+    #"table" now has three different tables i.e.
+    #I, IA and II.
+    attributes = list() 
+    
+    for item in table:
+        for cols in item.columns.values():
+            attributes.append(cols.name)
+    attributes = list(set(attributes))
+    #"attributes" is for the master table.
+    
+    #We are adding those columns in the tables
+    #that the tables don't have and initializing
+    #them with 'None'.
+    for item in table:
+        for attrs in attributes:
+            item_attr = [cols.name for cols in item.columns.values()]
+            if attrs not in item_attr:
+                new_column = Column(['-']*len(item), name=attrs, dtype='object_')
+                item.add_column(new_column)
+
+    #Just add a column for ID
+    Map = {0:'I', 1:'IA', 2:'II'}
```

Comment:
do not use `Map` as a variable name, there are already `map`s in python and sunpy. Are you sure you cannot use just a simple list: `srs_tables = ['I', 'IA', 'II']` ?